### PR TITLE
[case](regression-test) Add backup & restore regression test with auto Bucket and dynamic partition

### PR DESCRIPTION
**Pull request by Bears0haunt**
1.(case)Table with auto Bucket

   Table with dynamic partition
2. (case)set "reserve_dynamic_partition_enable = true" on restore properties 
3. (case)set "reserve_dynamic_partition_enable = false" on restore properties

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        ON (${tableName})
+    """    
+
+    while (!syncer.checkSnapshotFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    def snapshot = syncer.getSnapshotTimestamp(repoName, snapshotName)
+    assertTrue(snapshot != null)
+
+    sql "TRUNCATE TABLE ${dbName}.${tableName}"
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy:87]:
```suggestion
    sql "DROP TABLE ${dbName}.${tableName} FORCE"
```

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        (
+            "backup_timestamp" = "${snapshot}",
+            "reserve_dynamic_partition_enable" = "true",
+            "replication_num" = "1"
+        )
+    """
+    
+    // def restore_properties = sql "SHOW CREATE TABLE ${dbName}.${tableName}"
+
+    // assertTrue(restore_properties.indexOf("\"dynamic_partition.enable\" = \"true\"") != -1)
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy:103]: The result of `SHOW CREATE TABLE` contains a row with two columns, like: 

```
| Table                                 | Create Table 
| dynamic_partition_reserve_true_table  | CREATE TABLE XXXXX
```

To access the first row and second column, use `restore_properties[0][1]`

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_false.groovy [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        ON (${tableName})
+    """    
+
+    while (!syncer.checkSnapshotFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    def snapshot = syncer.getSnapshotTimestamp(repoName, snapshotName)
+    assertTrue(snapshot != null)
+
+    sql "TRUNCATE TABLE ${dbName}.${tableName}"
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_false.groovy:87]: DROP table to reset the properties.

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_false.groovy [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            "replication_num" = "1"
+        )
+    """
+    while (!syncer.checkAllRestoreFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    
+    // def restore_properties = sql "SHOW CREATE TABLE ${dbName}.${tableName}"
+
+    // assertTrue(restore_properties.indexOf("\"dynamic_partition.enable\" = \"false\"") != -1)
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_false.groovy:106]: ditto

File: regression-test/suites/backup_restore/test_backup_restore_table_auto_bucket.groovy [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        ON (${tableName})
+    """    
+
+    while (!syncer.checkSnapshotFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    def snapshot = syncer.getSnapshotTimestamp(repoName, snapshotName)
+    assertTrue(snapshot != null)
+
+    sql "TRUNCATE TABLE ${dbName}.${tableName}"
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_table_auto_bucket.groovy:61]: Consider dropping the table and verifying that `DISTRIBUTED BY HASH(`id`) BUCKETS AUTO` exists in the result of `SHOW CREATE TABLE` after restoring finished.

**Review by w41ter — COMMENTED** [review]
Reset LGTM

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    
+    while (!syncer.checkAllRestoreFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    result = sql "SELECT * FROM ${dbName}.${tableName}"
+    assertEquals(result.size(), 20);
+
+    def restore_properties = sql "SHOW CREATE TABLE ${dbName}.${tableName}"
+
+    assertTrue(restore_properties[0][1].indexOf("\"dynamic_partition.enable\" = \"true\"") != -1)
```
Comment (dataroaring) [inline regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy:109]: please check properties of dynamic_partition.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run buildall

**Comment by Bears0haunt** [comment]
run COMPILE

**Comment by Bears0haunt** [comment]
run compile

**Comment by Bears0haunt** [comment]
run compile(doris_compile)

**Comment by Bears0haunt** [comment]
run buildall

File: regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy (hunk 6 above)
Comment (JackDrogon) [reply regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy:109]: use contains to check， not indexOf

File: regression-test/suites/backup_restore/test_backup_restore_table_auto_bucket.groovy [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            "replication_num" = "1"
+        )
+    """
+    while (!syncer.checkAllRestoreFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+
+    def restore_properties = sql "SHOW CREATE TABLE ${dbName}.${tableName}"
+
+    assertTrue(restore_properties[0][1].indexOf("DISTRIBUTED BY HASH(`id`) BUCKETS AUTO") != -1)
```
Comment (JackDrogon) [inline regression-test/suites/backup_restore/test_backup_restore_table_auto_bucket.groovy:79]: use contains

**Review by JackDrogon — CHANGES_REQUESTED** [review]

**Comment by Bears0haunt** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**